### ISSUE TITLE
I installed this extension using composer , but Yii2 cannot find the Class when i try to use it , what is the problem?

### ISSUE THREAD

**Issue by zhexinhe** [yiisoft/yii2-imagine#8]
the problem is : 
![image](https://cloud.githubusercontent.com/assets/13096256/9380443/be74887c-4762-11e5-8534-6a85da46124c.png)
but the IDE can find this file and can open it 
![image](https://cloud.githubusercontent.com/assets/13096256/9380488/202dcf74-4763-11e5-95e1-41434f6441c8.png)

what is the problem ??

**Comment by cebe** [comment]
have you included the composer autoloader in your project? should be included in the index.php file.

**Comment by zhexinhe** [comment]
this is my index.php file , is there anything i have missed? or there is anything is worry ? 
![image](https://cloud.githubusercontent.com/assets/13096256/9546394/ebd95f8c-4dc4-11e5-8ab9-dd04bd45cbf5.png)

**Comment by cebe** [comment]
can you show the ouput of `composer show -i` should be a list of installed packages.

**Comment by zhexinhe** [comment]
i can see this package in my composer , but the issue is always here. what is my problem?

![default](https://cloud.githubusercontent.com/assets/13096256/10279013/ae15b2ca-6b94-11e5-84c7-59e23ad29045.jpg)

![2](https://cloud.githubusercontent.com/assets/13096256/10279060/1df50bb8-6b95-11e5-88fc-27c954a40c85.jpg)

**Comment by cebe** [comment]
the [imagine/imagine](https://github.com/yiisoft/yii2-imagine/blob/master/composer.json#L22) package seems to be not installed. you should try to run composer update again and see if there are any errors.

**Comment by empoower** [comment]
Hello! I have the same problem. The package is installed but yii autoloader can't find the class.
 Class Imagine\Gd\Imagine not found

**Comment by cebe** [comment]
@empoower please paste the complete output of `composer show -i`.

**Comment by empoower** [comment]
@cebe 
asofter/yii2-imperavi-redactor                dev-master 087ab0e Imperavi Redactor WYSIWYG widget (OEM-licensed for Yii 2)
bower-asset/ace-builds                        v1.1.9
bower-asset/admin-lte                         v2.3.0             Admin dashboard and control panel template
bower-asset/blueimp-canvas-to-blob            2.2.0              JavaScript Canvas to Blob is a function to convert canvas elements ...
bower-asset/blueimp-file-upload               9.7.2              File Upload widget with multiple file selection, drag&amp;drop supp...
bower-asset/blueimp-load-image                1.14.0             JavaScript Load Image is a library to load images provided as File ...
bower-asset/blueimp-tmpl                      2.5.5              < 1KB lightweight, fast & powerful JavaScript templating engine wit...
bower-asset/bootstrap                         v3.3.5             The most popular front-end framework for developing responsive, mob...
bower-asset/eonasdan-bootstrap-datetimepicker 4.17.37            bootstrap3 datetimepicker
bower-asset/flot                              v0.8.3
bower-asset/font-awesome                      v4.4.0             Font Awesome
bower-asset/html5shiv                         3.7.3
bower-asset/jquery                            2.1.4
bower-asset/jquery-slimscroll                 v1.3.6             slimScroll is a small jQuery plugin that transforms any div into a ...
bower-asset/jquery-ui                         1.11.4
bower-asset/jquery.inputmask                  3.1.63             jquery.inputmask is a jquery plugin which create an input mask.
bower-asset/moment                            2.10.6
bower-asset/moment-timezone                   0.4.0              Parse and display moments in any timezone.
bower-asset/punycode                          v1.3.2
bower-asset/typeahead.js                      v0.10.5
bower-asset/yii2-pjax                         v2.0.4
cebe/markdown                                 1.1.0              A super fast, highly extensible markdown parser for PHP
codeception/codeception                       2.1.3              BDD-style testing framework
doctrine/instantiator                         1.0.5              A small, lightweight utility to instantiate objects in PHP without ...
ezyang/htmlpurifier                           v4.6.0             Standards compliant HTML filter written in PHP
facebook/webdriver                            1.0.2              A PHP client for WebDriver
fzaninotto/faker                              v1.5.0             Faker is a PHP library that generates fake data for you.
guzzlehttp/guzzle                             6.1.0              Guzzle is a PHP HTTP client library
guzzlehttp/promises                           1.0.2              Guzzle promises library
guzzlehttp/psr7                               1.2.0              PSR-7 message implementation
imagine/imagine                               v0.5.0             Image processing for PHP 5.3
intervention/image                            2.3.2              Image handling and manipulation library with support for Laravel in...
kartik-v/bootstrap-fileinput                  v4.2.7             An enhanced HTML 5 file input for Bootstrap 3.x with features for f...
kartik-v/bootstrap-star-rating                v3.5.4             A simple yet powerful JQuery star rating plugin for Bootstrap.
kartik-v/dependent-dropdown                   v1.4.3             A multi level dependent dropdown JQuery plugin that allows nested d...
kartik-v/mpdf                                 v1.0.2             A PHP class to generate PDF files from HTML with Unicode/UTF-8 and ...
kartik-v/yii2-export                          dev-master 69b9fab A library to export server/db data in various formats (e.g. excel, ...
kartik-v/yii2-grid                            v3.0.7             Yii 2 GridView on steroids. Various enhancements and utilities for ...
kartik-v/yii2-krajee-base                     v1.7.7             Base library and foundation components for all Yii2 Krajee extensions.
kartik-v/yii2-money                           v1.2.1             An advanced money mask input for Yii 2.0 based on jquery maskmoney ...
kartik-v/yii2-mpdf                            dev-master f65fe85 A Yii2 wrapper component for the mPDF library which generates PDF f...
kartik-v/yii2-widget-activeform               v1.4.4             Enhanced Yii2 active-form and active-field with full bootstrap styl...
kartik-v/yii2-widget-affix                    v1.0.0             A scrollspy and affixed enhanced navigation to highlight page secti...
kartik-v/yii2-widget-alert                    v1.1.0             A widget to generate alert based notifications using bootstrap-aler...
kartik-v/yii2-widget-colorinput               v1.0.0             An enhanced Yii 2 widget encapsulating the HTML 5 color input (sub ...
kartik-v/yii2-widget-datepicker               v1.3.3             Enhanced Yii2 wrapper for the bootstrap datepicker plugin (sub repo...
kartik-v/yii2-widget-datetimepicker           v1.4.0             Enhanced Yii2 wrapper for the bootstrap datetimepicker plugin (sub ...
kartik-v/yii2-widget-depdrop                  v1.0.2             Widget that enables setting up dependent dropdowns with nested depe...
kartik-v/yii2-widget-fileinput                v1.0.3             An enhanced FileInput widget for Bootstrap 3.x with file preview, m...
kartik-v/yii2-widget-growl                    v1.1.1             A widget to generate growl based notifications using bootstrap-grow...
kartik-v/yii2-widget-rangeinput               v1.0.0             An enhanced Yii 2 widget encapsulating the HTML 5 range input (sub ...
kartik-v/yii2-widget-rating                   v1.0.0             A Yii2 widget for the simple yet powerful bootstrap-star-rating plu...
kartik-v/yii2-widget-select2                  v2.0.3             Enhanced Yii2 wrapper for the Select2 jQuery plugin (sub repo split...
kartik-v/yii2-widget-sidenav                  v1.0.0             An enhanced side navigation menu styled for bootstrap (sub repo spl...
kartik-v/yii2-widget-spinner                  v1.0.0             A widget to render animated CSS3 loading spinners with VML fallback...
kartik-v/yii2-widget-switchinput              v1.3.0             A Yii2 wrapper widget for the Bootstrap Switch plugin to use checkb...
kartik-v/yii2-widget-timepicker               v1.0.0             Enhanced Yii2 wrapper for the bootstrap timepicker plugin (sub repo...
kartik-v/yii2-widget-touchspin                v1.2.0             A Yii2 wrapper widget for the Bootstrap Switch plugin to use checkb...
kartik-v/yii2-widget-typeahead                v1.0.1             Enhanced Yii2 wrapper for the Twitter Typeahead plugin (sub repo sp...
kartik-v/yii2-widgets                         dev-master b7eb849 Collection of useful widgets for Yii Framework 2.0 extending functi...
league/flysystem                              1.0.15             Filesystem abstraction: Many filesystems, one API.
league/glide                                  0.3.5              Wonderfully easy on-demand image manipulation library with an HTTP ...
league/tactician                              v0.6.1             A small, flexible command bus. Handy for building service layers.
league/url                                    3.3.5              League/url is a lightweight PHP Url manipulating library
mihaildev/yii2-elfinder                       1.1.3              Yii2 ElFinder
phpdocumentor/reflection-docblock             2.0.4
phpmailer/phpmailer                           v5.2.6             PHPMailer is a full-featured email creation and transfer class for PHP
phpoffice/phpexcel                            1.8.1              PHPExcel - OpenXML - Read, Create and Write Spreadsheet documents i...
phpspec/php-diff                              v1.0.2             A comprehensive library for generating differences between two hash...
phpspec/prophecy                              v1.5.0             Highly opinionated mocking framework for PHP 5.3+
phpunit/php-code-coverage                     2.2.3              Library that provides collection, processing, and rendering functio...
phpunit/php-file-iterator                     1.4.1              FilterIterator implementation that filters files based on a list of...
phpunit/php-text-template                     1.2.1              Simple template engine.
phpunit/php-timer                             1.0.7              Utility class for timing
phpunit/php-token-stream                      1.4.8              Wrapper around PHP's tokenizer extension.
phpunit/phpunit                               4.8.10             The PHP Unit Testing framework.
phpunit/phpunit-mock-objects                  2.3.8              Mock Object library for PHPUnit
psr/http-message                              1.0                Common interface for HTTP messages
psr/log                                       1.0.0              Common interface for logging libraries
rmrevin/yii2-postman                          2.2.4              Mail module for Yii2.
sebastian/comparator                          1.2.0              Provides the functionality to compare PHP values for equality
sebastian/diff                                1.3.0              Diff implementation
sebastian/environment                         1.3.2              Provides functionality to handle HHVM/PHP environments
sebastian/exporter                            1.2.1              Provides the functionality to export PHP variables for visualization
sebastian/global-state                        1.0.0              Snapshotting of global state
sebastian/recursion-context                   1.0.1              Provides functionality to recursively process PHP variables
sebastian/version                             1.0.6              Library that helps with managing the version number of Git-hosted P...
swiftmailer/swiftmailer                       v5.4.1             Swiftmailer, free feature-rich PHP mailer
symfony/browser-kit                           v2.7.5             Symfony BrowserKit Component
symfony/console                               v2.7.5             Symfony Console Component
symfony/css-selector                          v2.7.5             Symfony CssSelector Component
symfony/debug                                 v2.7.5             Symfony Debug Component
symfony/dom-crawler                           v2.7.5             Symfony DomCrawler Component
symfony/event-dispatcher                      v2.7.5             Symfony EventDispatcher Component
symfony/finder                                v2.7.5             Symfony Finder Component
symfony/http-foundation                       v2.7.5             Symfony HttpFoundation Component
symfony/http-kernel                           v2.7.5             Symfony HttpKernel Component
symfony/yaml                                  v2.7.5             Symfony Yaml Component
trntv/cheatsheet                              0.1                Cheat sheet of knowledge you may use in your code
trntv/probe                                   0.2.1              System information provider
trntv/yii2-aceeditor                          2.0.0              Yii2 ajax.org Ace Editor widget
trntv/yii2-bootstrap-datetimepicker           1.5.0              Datetimepicker widget for Yii2 framework
trntv/yii2-file-kit                           1.1.5              Yii2 file upload and storage kit
trntv/yii2-glide                              1.0.0              Yii2 Glide Extension
trntv/yii2-tactician                          0.1.1              A simple, flexible command bus. This package provide it's integrati...
true/punycode                                 v2.0.1             A Bootstring encoding of Unicode for Internationalized Domain Names...
vlucas/phpdotenv                              v2.0.1             Loads environment variables from `.env` to `getenv()`, `$_ENV` and ...
yiisoft/yii2                                  dev-master 38a7f24 Yii PHP Framework Version 2
yiisoft/yii2-authclient                       2.0.5              External authentication via OAuth and OpenID for the Yii framework
yiisoft/yii2-bootstrap                        2.0.5              The Twitter Bootstrap extension for the Yii framework
yiisoft/yii2-codeception                      2.0.4              The Codeception integration for the Yii framework
yiisoft/yii2-composer                         2.0.3              The composer plugin for Yii extension installer
yiisoft/yii2-debug                            2.0.5              The debugger extension for the Yii framework
yiisoft/yii2-faker                            2.0.3              Fixture generator. The Faker integration for the Yii framework.
yiisoft/yii2-gii                              2.0.4              The Gii extension for the Yii framework
yiisoft/yii2-imagine                          2.0.3              The Imagine integration for the Yii framework
yiisoft/yii2-jui                              2.0.4              The Jquery UI extension for the Yii framework
yiisoft/yii2-swiftmailer                      2.0.4              The SwiftMailer integration for the Yii framework
zxbodya/yii2-gallery-manager                  dev-master 363f712 Extension for yii, that allows to manage image galleries

**Comment by cebe** [comment]
@empoower https://github.com/yiisoft/yii2-imagine/issues/8#issuecomment-133413702

**Comment by empoower** [comment]
Everything works fine now. I deleted all files in vendor and reload via composer.
